src/coap_net.c

```diff
@@ -3406,8 +3406,9 @@ handle_response(coap_context_t *context, coap_session_t *session,
 
   /* Call application-specific response handler when available. */
   if (context->response_handler) {
-    if (context->response_handler(session, sent, rcvd,
-                                  rcvd->mid) == COAP_RESPONSE_FAIL)
+    if ((context->response_handler(session, sent, rcvd,
+                                   rcvd->mid) == COAP_RESPONSE_FAIL)
+        && (rcvd->type != COAP_MESSAGE_ACK))
```

Comment:
My preference is to have the && at the end of the previous line (3410), which then means that the rest of line 3411 needs to be aligned under the opening (.